indy_node/test/nym_txn/test_get_nym_versions.py

```diff
@@ -0,0 +1,198 @@
+    )
+
+    update_ts = replies[0][1]["result"]["txnTime"]
+
+    # Get previous nym data by exact timestamp
+    get_nym_request = build_get_nym_request(did, did, timestamp)
+    request_couple = sdk_sign_and_send_prepared_request(
+        looper, sdk_wallet_endorser, sdk_pool_handle, get_nym_request
+    )
+    replies = sdk_get_and_check_replies(looper, [request_couple])
+
+    assert (
+        json.loads(replies[0][1]["result"]["data"])["diddocContent"] == diddoc_content_json
+    )
+
+    # Get previous nym data by timestamp but not exact
+    ts = randint(timestamp + 1, update_ts - 1)
+    get_nym_request = build_get_nym_request(did, did, ts)
+    request_couple = sdk_sign_and_send_prepared_request(
+        looper, sdk_wallet_endorser, sdk_pool_handle, get_nym_request
+    )
+    replies = sdk_get_and_check_replies(looper, [request_couple])
+
+    assert (
+        json.loads(replies[0][1]["result"]["data"])["diddocContent"] == diddoc_content_json
+    )
+
+
+def test_get_previous_nym_data_by_seq_no(
+    looper, sdk_pool_handle, sdk_wallet_endorser_factory, diddoc_content, diddoc_content_json
+):
+    sdk_wallet_endorser = sdk_wallet_endorser_factory(diddoc_content)
+    _, did = sdk_wallet_endorser
+
+    # Get current nym data
+    get_nym_request = build_get_nym_request(did, did)
+    request_couple = sdk_sign_and_send_prepared_request(
+        looper, sdk_wallet_endorser, sdk_pool_handle, get_nym_request
+    )
+    replies = sdk_get_and_check_replies(looper, [request_couple])
+
+    # Get seq_no from data
+    seq_no = replies[0][1]["result"]["seqNo"]
+
+    # Write new nym data
+    new_diddoc_content = copy.deepcopy(diddoc_content)
+    new_diddoc_content["serviceEndpoint"][0][
+        "serviceEndpoint"
+    ] = "https://new.example.com"
+    new_diddoc_content = json.dumps(new_diddoc_content)
+
+    time.sleep(3)
+
+    nym_request = build_nym_request(did, did, None, new_diddoc_content, None)
+    request_couple = sdk_sign_and_send_prepared_request(
+        looper, sdk_wallet_endorser, sdk_pool_handle, nym_request
+    )
+    sdk_get_and_check_replies(looper, [request_couple])
+
+    get_nym_request = build_get_nym_request(did, did)
+    request_couple = sdk_sign_and_send_prepared_request(
+        looper, sdk_wallet_endorser, sdk_pool_handle, get_nym_request
+    )
+    replies = sdk_get_and_check_replies(looper, [request_couple])
+
+    assert (
+        json.loads(replies[0][1]["result"]["data"])["diddocContent"] == new_diddoc_content
+    )
+
+    # Get previous nym data by seq_no
+    get_nym_request = build_get_nym_request(did, did, seq_no=seq_no)
+    request_couple = sdk_sign_and_send_prepared_request(
+        looper, sdk_wallet_endorser, sdk_pool_handle, get_nym_request
+    )
+    replies = sdk_get_and_check_replies(looper, [request_couple])
+
+    assert (
+        json.loads(replies[0][1]["result"]["data"])["diddocContent"] == diddoc_content_json
+    )
```

Comment:
These tests could use a reduction in copy-pasted txn posting and checking. This would improve readability of the tests.